PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalInternalClient.java

```diff
@@ -102,8 +109,20 @@ void sendRequest(final Context context, final PayPalRequest payPalRequest,
                                         }
 
                                         paymentAuthRequest
-                                            .clientMetadataId(clientMetadataId)
-                                            .approvalUrl(parsedRedirectUri.toString());
+                                                .clientMetadataId(clientMetadataId);
+
+                                        if (linkType.equals("universal")) {
+                                            if (pairingId != null && !pairingId.isEmpty()) {
+                                                paymentAuthRequest
+                                                        .approvalUrl(createAppSwitchUri(parsedRedirectUri).toString());
+                                            } else {
+                                                callback.onResult(null, new Exception("Missing BA Token for PayPal App Switch."));
```

Comment:
```suggestion
                                                callback.onResult(null, new BraintreeException("Missing BA Token for PayPal App Switch."));
```
`BraintreeException` is the standard exception type we use in the SDK when an error happens.